eCraft.appFactory.appFactoryService/Logger.cs

```diff
@@ -61,15 +63,31 @@ private static void AppendToLog(string logFileName, string message)
             // to file sequentially, but I opted for this extremely simplistic approach for now.
             lock (loggingLock)
             {
-                using (var fs = File.AppendText(logFileName))
+                try
+                {
+                    using (var fs = File.AppendText(logFileName))
+                    {
+                        var output = String.Format("{0} {1}", DateTime.UtcNow.ToString("o"), message);
+                        fs.WriteLine(output);
+                        Trace.WriteLine(output);
+                    }
+                }
+                catch (Exception e)
                 {
-                    var output = String.Format("{0} {1}", DateTime.UtcNow.ToString("o"), message);
-                    fs.WriteLine(output);
-                    Trace.WriteLine(output);
+                    WriteEventLog(e);
                 }
             }
         }
 
+        private static void WriteEventLog(Exception e)
+        {
+            if (!EventLog.SourceExists(EventSourceName))
+            {
+                EventLog.CreateEventSource(EventSourceName, "Application");

```

Comment:
does it have permission to create an event source?

Reply:
@sdahlbac the manifest says 
`<requestedExecutionLevel  level="requireAdministrator" uiAccess="false" />`

so it should be alright

Reply:
ok, let's hope so